res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,736 @@
+
+// Initialise and shutdown stuff.
+// ========================================================
+ReloopBeatmix24.TurnLEDsOff = function() {
+    // Turn all LEDS off
+    var i, j;
+    for (i = 0x91; i <= 0x94; i++) { // 4 decks
+
+        // Pads
+        for (j = 0x00; j <= 0x0F; j++) {
+            midi.sendShortMsg(i, j, OFF);
+            midi.sendShortMsg(i, j + SHIFT, OFF);
+        }
+        for (j = 0x10; j <= 0x1F; j++) {
+            midi.sendShortMsg(i, j, OFF);
+        }
+
+        // Play/cue/cup/sync/pitch bend
+        for (j = 0x20; j <= 0x25; j++) {
+            midi.sendShortMsg(i, j, OFF);
+            midi.sendShortMsg(i, j + SHIFT, OFF);
+        }
+
+        // Jog Wheel leds
+        ReloopBeatmix24.AllJogLEDsToggle(i, OFF);
+
+        // Load + Cue
+        midi.sendShortMsg(i, 0x50, OFF);
+        midi.sendShortMsg(i, 0x52, OFF);
+        midi.sendShortMsg(i, 0x72, OFF);
+    }
+};
+
+ReloopBeatmix24.connectControls = function() {
+    var group;
+
+    // Channels controls
+    for (var i = 1; i <= 4; i++) {
+        group = "[Channel" + i + "]";
+        engine.connectControl(group, "track_samples",
+            "ReloopBeatmix24.deckLoaded");
+        engine.connectControl(group, "play",
+            "ReloopBeatmix24.ChannelPlay");
+        engine.connectControl(group, "playposition",
+            "ReloopBeatmix24.JogLed");
+        engine.connectControl(group, "loop_end_position",
+            "ReloopBeatmix24.loopDefined");
+        engine.softTakeover(group, "rate", true);
+        engine.setValue("[EffectRack1_EffectUnit1]",
+            "group_" + group + "_enable", 0);
+        engine.setValue("[EffectRack1_EffectUnit2]",
+            "group_" + group + "_enable", 0);
+        channelPlaying[group] = engine.getValue(group, "play") ? true : false;
+        JogBlinking[group] = false;
+    }
+
+    // Samplers controls
+    for (i = 1; i <= 8; i++) {
+        group = "[Sampler" + i + "]";
+        engine.connectControl(group, "track_samples",
+            "ReloopBeatmix24.deckLoaded");
+        engine.connectControl(group, "play",
+            "ReloopBeatmix24.SamplerPlay");
+    }
+
+    // Effects reset
+    engine.setValue("[EffectRack1_EffectUnit1]", "group_[Master]_enable", 0);
+    engine.setValue("[EffectRack1_EffectUnit2]", "group_[Master]_enable", 0);
+};
+
+ReloopBeatmix24.init = function(id, debug) {
+    ReloopBeatmix24.id = id;
+    ReloopBeatmix24.TurnLEDsOff(); // Turn off all LEDs
+    ReloopBeatmix24.connectControls(false);
+
+    for (var i = 1; i <= 4; i++) {
+        engine.trigger("[Channel" + i + "]", "loop_end_position");
+    }
+
+    print("Reloop Beatmix: " + id + " initialized.");
+};
+
+ReloopBeatmix24.shutdown = function() {
+    ReloopBeatmix24.TurnLEDsOff(); // Turn off all LEDs
+    print("Reloop Beatmix: " + ReloopBeatmix24.id + " shut down.");
+};
+
+// Button functions.
+// ========================================================
+ReloopBeatmix24.GetNextRange = function(OV) {
```

Comment:
Does "OV" mean "octave"? If so, I think using "octave" for the variable name would be a little clearer.